src/Sources/AbstractSource.php

```diff
@@ -135,44 +161,100 @@ function ($migration_class) {
         $this->onUnInstall();
     }
 
-    protected function save(Model $me, Model $primary)
+    /**
+     * Checks for new primary entity and source info, updated source info, and updates the audit table as needed.
+     *
+     * @param Model $source
+     * @param $primary
+     * @param $existing_source
+     * @param array $sourceColumns
+     * @return void
+     */
+    protected function save(Model $source, $primary, $existing_source, array $sourceColumns)
     {
-        // @todo: update $me name
-        // @todo: Only save model if something has changed
-        // @todo: in the Audit, describe what changed
-        // Now, we save the primary version
-        $primary->save();
+        // Create new primary entity and source if game doesn't already exist.
+        if ($primary !== null) {
+            $primary->save();
+            $column_name = static::getPrimaryEntityJoinColumnName();
+            $source->{$column_name} = $primary->id;
+            $source->save();
+        }
 
-        //@todo: update game_id to more generic
-        $me->game_id = $primary->id;
+        // Update source if changes are found.
+        if ($existing_source !== null) {
+            $changes = $this->findChanges($source, $existing_source, $sourceColumns);
+            $source->id = $existing_source->id;
+            $existing_source->save();
+        } else {
+            $changes = null;
```

Comment:
You can set this to be an empty array here `[]` and then remove the "if not null" check later on. Looping through an empty array does nothing.

Reply:
If you mean set `$changes  = []` the reason for that is I needed a way to tell the difference between a new bgg instance and one that's already created with no changes.

Reply:
Ah, good point